### PR TITLE
Logserver policy and storage solution proposal

### PR DESCRIPTION
**Pull request by diabelko**

### CODE REVIEW

File: source/rfc/logs.rst [hunk 1]
```diff
@@ -1,3 +1,52 @@
 Logs
 ====
 
+Problem
+-------
+
+Our log server is full way too often. We're currently running on a 14-day policy, even though we should be keeping at least 30 days of logs.
```
Comment (wurbanski) [inline source/rfc/logs.rst]: ~Our~The

File: source/rfc/logs.rst [hunk 2]
```diff
@@ -1,3 +1,52 @@
...
+
+Nightlies
+~~~~~~~~~
+
+Nightly logs should follow the very same policy that build artifacts (packages and containers) will follow to ensure that we have all information we need historically. What's more, we should consider keeping nightly logs for a longer period of time (60 or 90 days), as someone might use an older build (even though we don't provide any support for them).
+
+Solution
+--------
+
+We should increase server size to 700GB (allowing some wiggle room for other machines to expand if necessary) which should last us during quiet time (20GB/day gives us 600GB with 30 days). In the meantime we should focus on a proper fix.
```
Comment (kravvcuCL) [inline source/rfc/logs.rst]: 600GB is the optimistic scenario, above you provided a range of 20-40GB/day. What's the distribution?

File: source/rfc/logs.rst [hunk 3]
```diff
@@ -1,3 +1,52 @@
...
+Notes:
+
+- there's only 600GB free on the hypervisor
+- 2nd hypervisor has 800GB free
+- we seem to be generating from 20 to 40GB of logs per day (as of 2018-05-04)
+
+Retention policy
+----------------
+
+We should keep 30 days of logs for reviews - that should be enough to be able to merge properly a review.
```
Comment (wurbanski) [inline source/rfc/logs.rst]: merge properly a review -> properly merge a review.

File: source/rfc/logs.rst [hunk 4]
```diff
@@ -1,3 +1,52 @@
...
+
+Nightlies
+~~~~~~~~~
+
+Nightly logs should follow the very same policy that build artifacts (packages and containers) will follow to ensure that we have all information we need historically. What's more, we should consider keeping nightly logs for a longer period of time (60 or 90 days), as someone might use an older build (even though we don't provide any support for them).
```
Comment (wurbanski) [inline source/rfc/logs.rst:32]: > as someone might use an older build (even though we don't provide any support for them).

I think we could keep only parts of the logs in this case - e.g. sanity results

File: source/rfc/logs.rst [hunk 5]
```diff
@@ -1,3 +1,52 @@
...
+
+Storage
+~~~~~~~
+
+We should move to an object storage solution as soon as possible. One possibility would be AWS with S3. Having object storage mounted as fs allows us to store the logs same as before, but with no limits other than our own enforced policies.
+
+Nightly handling
+~~~~~~~~~~~~~~~~
+
+There should be a tool (or part of the artifact publisher) allowing us to select logs based on build number and store them separately to prevent 30-day policy from kicking in.
```
Comment (wurbanski) [inline source/rfc/logs.rst]: Please define the process (which builds, why, how to automate that), not the tool for this.

**Review by wurbanski — CHANGES_REQUESTED** [review]

File: source/rfc/logs.rst (hunk 4 above)
Comment (diabelko) [reply source/rfc/logs.rst:32]: We don't store sanity results on our end. Also, the size of those logs would be really minimal (especially compared to test logs), as we only have build, container and publishing logs there.

What is worth considering IMO is how to handle the logs from builds marked as monthly, as we need to keep the last 12 (full year).

File: source/rfc/logs.rst (hunk 2 above)
Comment (diabelko) [reply source/rfc/logs.rst]: Indeed, it is an optimistic one. To my defense, we actually have a quiet time (post-release) right now and this is supposed to be only temporary to make it less painful while we work on a proper fix.

Also, after some discussion decided to bump it up to 800GB, which might be even safer.

File: source/rfc/logs.rst [hunk 6]
```diff
@@ -1,3 +1,57 @@
...
+----------
+
+Storage
+~~~~~~~
+
+Because it's extremely difficult to properly estimate the log growth due to variety of possible reasons (amount of jobs, reviews, flakiness), We should move to an object storage solution as soon as possible. 
+
+One possibility would be AWS with S3. Having object storage mounted as fs (s3fs) allows us to store the logs same as before, but with no limits other than our own enforced policies as the storage is pretty much unlimited.
+
+Log curator
```
Comment (kravvcuCL) [inline source/rfc/logs.rst]: Maybe it's worth to mention that S3 buckets also have a retention policy mechanism we could utilise?

**Review by kravvcuCL — APPROVED** [review]

**Review by wurbanski — DISMISSED** [review]

File: source/rfc/logs.rst (hunk 6 above)
Comment (diabelko) [reply source/rfc/logs.rst]: Added the info. Good catch.

**Comment by kravvcuCL** [comment]
We can add that we could introduce terraform / IoC into the project, should we use S3. Using terraform is kinda :feelsgood:. TAL 'Using object lifecycle' https://www.terraform.io/docs/providers/aws/r/s3_bucket.html

**Review by kravvcuCL — APPROVED** [review]

**Pull request merged**